Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -502,6 +527,48 @@ theorem X_dvd_iff {φ : R⟦X⟧} : (X : R⟦X⟧) ∣ φ ↔ constantCoeff R φ
   · intro m hm
     rwa [Nat.eq_zero_of_le_zero (Nat.le_of_succ_le_succ hm)]
 
+theorem eq_zero_or_eq_zero_of_mul_eq_zero [NoZeroDivisors R] (φ ψ : R⟦X⟧) (h : φ * ψ = 0) :
```

Comment:
What's the advantage of having this lemma, rather than simply the instance below? If we keep both, I'd add a line of doc to this lemma saying that the instance `NoZeroDivisors` is there.

Reply:
I put it in the instance…